### ISSUE TITLE
Deploy failed, error: failed to unwrap Info.plist from ipa, error: multiple file generated

### ISSUE THREAD

**Issue by patrickbussmann** [bitrise-io/steps-deploy-to-bitrise-io#34]
Hey @godrei,

I have an issue with 1.3.0 and 1.3.1 of this version. If I'm using 1.2.9 it's working well :+1: 

	Uploading ipa file: /Users/vagrant/deploy/example.ipa
	analyzing ipa
	Deploy failed, error: failed to unwrap Info.plist from ipa, error: multiple file generated

And screenshot.
![Bitrise example build log screenshot](https://cloud.githubusercontent.com/assets/15617021/26475382/dbb34cba-41b7-11e7-93eb-00ab168c9e05.png)

Don't know why it is unable to unwrap the file now because in 1.2.9 it's working fine 🙂

Can you check it please? Or I will do it later if I have more time. There are some apps waiting for deploy 😎

**Comment by godrei** [comment]
Hi @patrickbussmann ,
thanks for the report i will fix this ASAP!

**Comment by viktorbenei** [comment]
It might be obvious, but just to be sure: while we're working on the fix **you can use the `1.2.9` version of the step**. That's part of why we have step versions and why we try our best to keep compatibility with older versions, so that you can use an older version as long as you want.

We'll fix the issue ASAP, just wanted to highlight this as a workaround in the meantime, to get those "ASAP deploys" working ;)

**Comment by patrickbussmann** [comment]
Yes of course, @viktorbenei :+1: 
But I thought it's good for test reasons if I remove the version number from the steps so that I always have the newest version and if there are problems I can help to fix 😊

In this case I will stick to 1.2.9 but if I would have more time I would try to fix this. But currently there are so many updates in my apps so that I don't have time 😔

Thank's @godrei for checking. You must not fix this asap 😉

**Comment by viktorbenei** [comment]
Sure @patrickbussmann I definitely agree, this should only be a temporary solution for now, while we're working on the fix ;)

To be honest this 1.3.0 release was quite a significant rewrite, that's what causes the issues now even though we tried our best to test it properly. It's not common that we do a full rewrite like this, and we'll improve the tests of this step too :)

**Comment by godrei** [comment]
@patrickbussmann thanks again for the report!

We released a new version [1.3.2](https://github.com/bitrise-io/steps-deploy-to-bitrise-io/releases/tag/1.3.2) with fix for this issue.

I'll close this issue, but if you need more info/help just let us know and we'll reopen it!

**Comment by FranMartinRomero** [comment]
We still are having this issue with 1.3.2.

+------------------------------------------------------------------------------+
| (0) deploy-to-bitrise-io                                                     |
+------------------------------------------------------------------------------+
| id: deploy-to-bitrise-io                                                     |
| version: 1.3.2                                                               |
| collection: https://github.com/bitrise-io/bitrise-steplib.git                |
| toolkit: go                                                                  |
| time: 2017-05-30T05:26:41-07:00  

> Deploy failed, error: failed to unwrap Info.plist from ipa, error: command: (/usr/bin/unzip "XXX_Dev.ipa" "-d" "/var/folders/90/5stft2v13fb_m_gv3c8x9nwc0000gn/T/__ipa__420390992") failed, output: unzip:  cannot find or open XXX_Dev.ipa, XXX_Dev.ipa.zip or XXX_Dev.ipa.ZIP., error: exit status 9

It's only working with 1.2.9

**Comment by patrickbussmann** [comment]
@FranMartinRomero please ensure that you set the `BITRISE_IPA_PATH` in xcode-archive step or where you build with.

On my side it's working with 1.3.2 now.
The IPA Path is set on my side through `xcode-archive` step.
In Log you can see:

	Exporting outputs...

	The xcarchive path is now available in the Environment Variable: BITRISE_XCARCHIVE_PATH (value: /var/folders/90/5stft2v13fb_m_gv3c8x9nwc0000gn/T/__archive__048760614/Example-App.xcarchive)

	The app directory is now available in the Environment Variable: BITRISE_APP_DIR_PATH (value: /Users/vagrant/deploy/Example-App.app)

	The ipa path is now available in the Environment Variable: BITRISE_IPA_PATH (value: /Users/vagrant/deploy/Example-App.ipa)

	The dSYM dir path is now available in the Environment Variable: BITRISE_DSYM_DIR_PATH (value: /var/folders/90/5stft2v13fb_m_gv3c8x9nwc0000gn/T/__dsyms__213161800)
	The dSYM zip path is now available in the Environment Variable: BITRISE_DSYM_PATH (value: /Users/vagrant/deploy/Example-App.dSYM.zip)
	|                                                                              |
	+---+---------------------------------------------------------------+----------+
	| ✓ | xcode-archive                                                 | 104 sec  |
	+---+---------------------------------------------------------------+----------+

On your side you have a problem because the IPA is not found.
So normally it should'nt work with 1.2.9, too.
If you say its working with 1.2.9 then it's really interesting 😨
Because we talking about `unzip` which can't find your IPA.

Can you please add a script step **before** deploy to bitrise step which contains:
`ls -al`.
After you build it you'll see a list of all files in the current folder.
Normally you should see your `XXX_Dev.ipa` there. If not, please check that you're using the correct path 😉

**Comment by viktorbenei** [comment]
Thanks @patrickbussmann for the advise, but in this case I don't think the issue is related to a config, but rather to the 1.3.x revision (re-write) of this step.

Thanks for the report @FranMartinRomero !

@godrei should we move this into a separate github issue, or reopen this one?

**Comment by godrei** [comment]
Hi @FranMartinRomero , thanks for the report!

@viktorbenei the issue is related to the same package (go-utils/ziputil) but the source of the issue is a new one. 

@FranMartinRomero may i ask you for some more log? It would be useful to see the `List of files to deploy` section of the step log.

**Comment by viktorbenei** [comment]
If possible, @FranMartinRomero please share the whole step log, from the step header to the footer (just this step's)

**Comment by FranMartinRomero** [comment]
Sure! This is the List of files to deploy section

Switching to workflow: Internal_Deploy_App

+------------------------------------------------------------------------------+
| (0) deploy-to-bitrise-io                                                     |
+------------------------------------------------------------------------------+
| id: deploy-to-bitrise-io                                                     |
| version: 1.3.2                                                               |
| collection: https://github.com/bitrise-io/bitrise-steplib.git                |
| toolkit: go                                                                  |
| time: 2017-05-30T05:26:41-07:00                                              |
+------------------------------------------------------------------------------+
|                                                                              |

Configs:
- BuildURL: https://www.bitrise.io/build/dbb6c145c0a3f151
- APIToken: DolAVNziF4oLyVXF6rsQVQ
- IsCompress: false
- DeployPath: XXX_Dev.ipa
- NotifyUserGroups: none
- NotifyEmailList: 
- IsPublicPageEnabled: true

Deploying single file

List of files to deploy
- XXX_Dev.ipa

Deploying files

Uploading ipa file: XXX_Dev.ipa
analyzing ipa
Deploy failed, error: failed to unwrap Info.plist from ipa, error: command: (/usr/bin/unzip "XXX_Dev.ipa" "-d" "/var/folders/90/5stft2v13fb_m_gv3c8x9nwc0000gn/T/__ipa__420390992") failed, output: unzip:  cannot find or open XXX_Dev.ipa, XXX_Dev.ipa.zip or XXX_Dev.ipa.ZIP., error: exit status 9
|                                                                              |
+---+---------------------------------------------------------------+----------+
| x | deploy-to-bitrise-io (exit code: 1)                           | 7.62 sec |
+---+---------------------------------------------------------------+----------+
| Issue tracker: ...://github.com/bitrise-io/steps-deploy-to-bitrise-io/issues |
| Source: https://github.com/bitrise-io/steps-deploy-to-bitrise-io             |
+---+---------------------------------------------------------------+----------+

**Comment by viktorbenei** [comment]
Can it be that the file specified for the deploy actually does not exist at that path? (at the path specified at `DeployPath: ...`

**Comment by FranMartinRomero** [comment]
I can see a difference in the deploy_path between 1.2.9 and 1.3.2.
1.2.9 use deploy_path: /Users/vagrant/git/XXX/XXX_Dev.ipa
1.3.2.use XXX_Dev.ipa

**Comment by viktorbenei** [comment]
That will be the issue! Thanks @FranMartinRomero for reporting and for the infos!

@godrei pls schedule a fix for this

@FranMartinRomero in the meantime please use the 1.2.x version of this step, will let you know here once the issue is fixed

**Comment by FranMartinRomero** [comment]
Thanks @viktorbenei for your quick response. :)

**Comment by viktorbenei** [comment]
@godrei it seems that a sub dir is missing from the path, maybe some kind of path normalization issue (e.g. that the file's name is used instead of its path somewhere)

**Comment by viktorbenei** [comment]
@FranMartinRomero any time, and sorry for the inconvenience

**Comment by godrei** [comment]
Hi @FranMartinRomero there was an issue in our zip package, which only happened if we tried to use the package with relative paths. 
We released a new version of the step ([v1.3.3](https://github.com/bitrise-io/steps-deploy-to-bitrise-io/releases/tag/1.3.3)), whith the fixed zip package and from now the step expands the `deploy_path`.

@patrickbussmann & @viktorbenei thanks for the infos and clarifications!

I'll close this issue, but if someone needs more info/help just let us know and we'll reopen it!
